examples/memory_with_compression/_memory_storage.py

```diff
@@ -0,0 +1,80 @@
+# -*- coding: utf-8 -*-
+"""The memory storage base class."""
+from abc import abstractmethod
+from typing import Any, Iterable, Union
+
+from agentscope.message import Msg
+
+
+class MessageStorageBase:
```

Comment:
The base class is missing lifecycle management methods. Since async setup cannot happen in `__init__`, consider adding async hooks like `connect()`, `close()`, and `check_health()` for consistent resource management and status checking across storage backends.